src/nn/ops/reduce.ts

```diff
@@ -31,12 +32,28 @@ abstract class Reduce extends SingleOutputOperation {
     } else {
       this.keepDimensions_ = false;
     }
+
+    this.createOutput();
   }
 
   inputs(): MLOperand[] {
     return [this.input_];
   }
 
+  createOutput(): void {
+    const inpAxes = this.axes_ ?? [...Array(this.input_.rank()).keys()];
+    this.outputShape_ = this.input_.shape().slice();
+    for (let i = 0; i < inpAxes.length; ++i) {
+      this.outputShape_[inpAxes[i]] = 1;
+    }
+    if (!this.keepDimensions_) {
+      this.outputShape_ = this.outputShape_.filter((dim, axis) =>
+        !(dim === 1 && inpAxes.indexOf(axis) !== -1));
```

Comment:
Should `inpAxes.indexOf(axis) !== -1` always hold true?

Reply:
Yes, it's. Now the axis value is of `unsigned long` type.